aten/src/ATen/native/mps/operations/ScanKernel.mm

```diff
@@ -140,14 +141,165 @@ static void scan_mps_impl(const Tensor& self,
   });
 }
 
+// Utility function to get 2D grid dimensions for dispatch
+static std::pair<uint32_t, uint32_t> get_2d_grid_dims(const IntArrayRef& shape, const int64_t dim) {
+  size_t grid_x = 1;
+  size_t grid_y = 1;
+
+  for (int i = 0; i < dim; ++i) {
```

Comment:
If your input is int64_t, iterating over int32 can potentially create an infinite loop
```suggestion
  for (const auto  i : c10::irange(dim)) {
```